src/importer.ts

```diff
@@ -0,0 +1,79 @@
+import {EventMessage} from "./models";
+import {PapiClient} from "./papi-client";
+import {URL} from "./types";
+import * as Promise from "bluebird";
+
+export interface AmqpQueues {
+    event: string;
+}
+
+export interface AmqpConfig {
+    host: URL;
+    queues: AmqpQueues;
+}
+
+export interface ImporterConfig {
+    amqp: AmqpConfig;
+    maxBatchSize: number;
+    maxBatchWaitTimeMs: number;
+    delayBetweenBatchMs: number;
+}
+
+export class Importer {
+
+    constructor(
+        private papiClient: PapiClient,
+        private config: ImporterConfig,
+        private amqpConnectionProvider
+    ) {}
+
+    public importAllEvents(): void {
+        let runCount = 0;
+        this.amqpConnectionProvider
+            .flatMap(connection => connection.createChannel())
```

Comment:
That's how Observable works actually. createChannel returns an observable "stream" (onNext, .. onNext, onComplete) even for single observables (onNext and onComplete only).
Hence using flatMap is used to chain observables items (onComplete->onNext), as a way to serialize/sequence events.